sqs-fifo-delayed-queue-dynamodb/README.md

```diff
@@ -0,0 +1,58 @@
+# Amazon SQS FIFO Queue with delay using AWS Lambda and AWS DynamoDB
+
+This pattern allows introduction of a delay between processing of messages while maintaining order from individual client. Then sending it sequentially to the downstream for processing, to minimize the consequences of unordered events.
+
+Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/sqs-fifo-delayed-queue-dynamodb
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ``` 
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd sqs-fifo-delayed-queue-dynamodb
+    ```
+1. From the command line, use AWS CDK to deploy the AWS resources for the pattern as specified in the `delay_fifo_queue_test/delay_fifo_queue_test_stack.py` file.
+    ```
+    python3 -m pip install -r requirements.txt
+    cdk synth
+    cdk deploy
+    ```
+
+## How it works
+
+This pattern deploys an Amazon SQS FIFO queue called `primary_queue`, a AWS Lambda function `process_queue_function`, a DynamoDB table `customer_table` and a second SQS FIFO queue `downstream_queue`. 
+
+When a messages from `primary_queue` is processed by the `process_queue_function`, it is checked against `customer_table` to see if another message from the same message sender has been processed with in a specified time frame.
+If true, the message is not processed and with be retried after the visibility timeout on the `primary_queue`.
+If false, the message is sent to the `downstream_queue` for processing. An entry is made to `customer_table` with a TTL.
+
+
+## Testing
+
+Edit lines 3 and 7 of `send_messages.sh` with the URL of your `primary_queue`. Run this script to send test messages to the queue.
```

Comment:
Where would I get the value for ```primary_queue```? 
I had to check the CloudFormation stack events to get the queue URL. There were 3 queues with different names, none mentioning primary_queue so it was not very clear. 

I was able to find the right queue and test it, however I would strongly use of CloudFormation Stack Outputs to output the required resources and mention that in test steps so it is easier for a customer to understand and follow steps.